mavros/src/plugins/hil.cpp

```diff
@@ -134,9 +134,9 @@ class HilPlugin : public plugin::PluginBase {
 		state_quat.rollspeed = req->imu.angular_velocity.x;
 		state_quat.pitchspeed = req->imu.angular_velocity.y;
 		state_quat.yawspeed = req->imu.angular_velocity.z;
-		state_quat.lat = req->geo.latitude;	// @warning geographic_msgs/GeoPoint.msg uses WGS 84 reference ellipsoid
-		state_quat.lon = req->geo.longitude;	// @TODO: Convert to AMSL to be received by the FCU
-		state_quat.alt = req->geo.altitude;	// related to issue #529
+		state_quat.lat = (int32_t)req->geo.latitude;	// @warning geographic_msgs/GeoPoint.msg uses WGS 84 reference ellipsoid
+		state_quat.lon = (int32_t)req->geo.longitude;	// @TODO: Convert to AMSL to be received by the FCU
+		state_quat.alt = (int32_t)req->geo.altitude;	// related to issue #529
```

Comment:
WHAT are you doing? That is not how you should convert int32 coord to double (or back)!